### PR TITLE
[CLI IMPROVEMENT]  improved cli message for new cmd

### LINKED ISSUE #43428: [Improvement]: Improve the CLI command outputs
### Description

Watched a [video tutorial on YT](https://www.youtube.com/watch?v=rhqPjz057zw) and noticed some hiccups in the CLI user experience. IMO, the CLI help text can be more intuitive in guiding the user throughout the build process. For example, we can add a `What's next` as the last part of the command output. This has already been added for the `bal build --cloud=docker`. We should update the other commands also as follows:

<meta charset="utf-8"><div dir="ltr" style="margin-left:0pt;" align="left">
Command | What’s Next
-- | --
`bal new <package-name>`</br></br>`bal new <package-name> -t main` | Execute `bal run <package-name>` to run the program</br></br>Execute `bal build <package-name>` to build the executable JAR
`bal new <package-name> -t service` | Execute `bal run <package-name>` to run the service</br></br>Execute `bal build <package-name>` to build the executable JAR
`bal new <package-name> -t lib` | Execute `bal pack <package-name>` to create the Ballerina archive
`bal build` | Execute `java -jar <relative-path-to-the-executable-file>` to run the program
`bal pack` | Execute `bal push` to publish the package to Ballerina Central.

</div></b>

### Describe your problem(s)

_No response_

### Describe your solution(s)

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by tanbirali**
Fixes #43428 

This is a Draft PR to understand if I am heading in the right direction.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by azinneera** [comment]
@tanbirali Thank you for the contribution. The changes you've made so far look great. Can you also address the rest of the changes to the `bal new` command, `bal build` and `bal pack` commands as mentioned in the issue? You will also have to fix the [test cases](https://github.com/ballerina-platform/ballerina-lang/tree/master/cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd) related to these commands.

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java [hunk 1]
```diff
@@ -304,6 +304,13 @@ public void execute() {
                 .build();
 
         taskExecutor.executeTasks(project);
+        Path targetDirPath = project.sourceRoot().resolve("target");
+        String packageName = project.sourceRoot().getFileName().toString();
+        String jarFileName = packageName+".jar"; 
+    Path jarFilePath = targetDirPath.resolve("bin").resolve(jarFileName);
+        outStream.println("What's next?");
+        outStream.println("Execute java -jar "+ jarFilePath.toString +" to run the program");
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java:312]:
```suggestion
        outStream.println("Execute java -jar " + jarFilePath.toString + " to run the program");
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PackCommand.java [hunk 2]
```diff
@@ -174,6 +174,8 @@ public void execute() {
             }
         } catch (ProjectException e) {
             CommandUtil.printError(this.errStream, e.getMessage(), null, false);
+            
+
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PackCommand.java:178]:
```suggestion
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java [hunk 3]
```diff
@@ -304,6 +304,13 @@ public void execute() {
                 .build();
 
         taskExecutor.executeTasks(project);
+        Path targetDirPath = project.sourceRoot().resolve("target");
+        String packageName = project.sourceRoot().getFileName().toString();
+        String jarFileName = packageName+".jar"; 
+    Path jarFilePath = targetDirPath.resolve("bin").resolve(jarFileName);
+        outStream.println("What's next?");
+        outStream.println("Execute java -jar " + jarFilePath.toString + " to run the program");
```
Comment (gimantha) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java:312]:
```suggestion
        outStream.println("Execute java -jar " + jarFilePath.toString() + " to run the program");
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java [hunk 4]
```diff
@@ -304,6 +304,13 @@ public void execute() {
                 .build();
 
         taskExecutor.executeTasks(project);
+        Path targetDirPath = project.sourceRoot().resolve("target");
+        String packageName = project.sourceRoot().getFileName().toString();
+        String jarFileName = packageName+".jar";
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java:309]:
```suggestion
        String jarFileName = packageName + ".jar"; 
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java [hunk 5]
```diff
@@ -304,6 +304,13 @@ public void execute() {
                 .build();
 
         taskExecutor.executeTasks(project);
+        Path targetDirPath = project.sourceRoot().resolve("target");
+        String packageName = project.sourceRoot().getFileName().toString();
+        String jarFileName = packageName+".jar"; 
+    Path jarFilePath = targetDirPath.resolve("bin").resolve(jarFileName);
+        outStream.println("What's next?");
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java:311]:
```suggestion
        outStream.println("\nWhat's next?");
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java [hunk 6]
```diff
@@ -304,6 +304,13 @@ public void execute() {
...
+        outStream.println("Execute java -jar " + jarFilePath.toString() + " to run the program");
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java:312]:
```suggestion
        outStream.println("\tExecute java -jar " + jarFilePath.toString() + " to run the program");
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java [hunk 7]
```diff
@@ -305,6 +305,31 @@ public void execute() {
                 packagePath = Paths.get(argList.get(0));
             }
             errStream.println("Created new package '" + packageName + "' at " + packagePath + ".");
+            errStream.println("What's next");
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java:308]:
```suggestion
            errStream.println("\nWhat's next");
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java [hunk 8]
```diff
@@ -305,6 +305,31 @@ public void execute() {
...
+             switch (template) {
+                case "main":
+                    errStream.println("Execute the below command to build the executable JAR file:");
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java:311]:
```suggestion
                    errStream.println("\tExecute the below command to build the executable JAR file:");
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java [hunk 9]
```diff
@@ -305,6 +305,31 @@ public void execute() {
...
+                    errStream.println("\t bal build " + packageName);
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java:312]:
```suggestion
                    errStream.println("\tbal build " + packageName);
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java [hunk 10]
```diff
@@ -305,6 +305,31 @@ public void execute() {
...
+                    break;
+
+                case "service":
+                    errStream.println("Execute the below command to run the service:");
+                    errStream.println("\t bal run " + packageName);
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java:317]:
```suggestion
                    errStream.println("\tbal run " + packageName);
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java [hunk 11]
```diff
@@ -305,6 +305,31 @@ public void execute() {
...
             errStream.println("Created new package '" + packageName + "' at " + packagePath + ".");
+            errStream.println("What's next");
+             switch (template) {
+                case "main":
+                    errStream.println("Execute the below command to build the executable JAR file:");
+                    errStream.println("\t bal build " + packageName);
+                    break;
+
+                case "service":
+                    errStream.println("Execute the below command to run the service:");
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java:316]:
```suggestion
                    errStream.println("\tExecute the below command to run the service:");
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java [hunk 12]
```diff
@@ -305,6 +305,31 @@ public void execute() {
...
+                    errStream.println("\t bal run " + packageName);
+                    errStream.println("Else, execute the below command to build the executable JAR file:");
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java:318]:
```suggestion
                    errStream.println("\tElse, execute the below command to build the executable JAR file:");
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java [hunk 13]
```diff
@@ -305,6 +305,31 @@ public void execute() {
...
+                    errStream.println("\t bal build " + packageName);
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java:319]:
```suggestion
                    errStream.println("\tbal build " + packageName);
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java [hunk 14]
```diff
@@ -305,6 +305,31 @@ public void execute() {
...
+                    break;
+
+                case "lib":
+                    errStream.println("Execute the below command to create the Ballerina archive:");
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java:323]:
```suggestion
                    errStream.println("\tExecute the below command to create the Ballerina archive:");
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java [hunk 15]
```diff
@@ -305,6 +305,31 @@ public void execute() {
...
+                    errStream.println("\t bal pack " + packageName);
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java:324]:
```suggestion
                    errStream.println("\tbal pack " + packageName);
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java [hunk 16]
```diff
@@ -305,6 +305,31 @@ public void execute() {
...
+                    break;
+
+                default:
+                    errStream.println("Execute the below command to run the program:");
+                    errStream.println("\t bal run " + packageName);
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java:329]:
```suggestion
                    errStream.println("\tbal run " + packageName);
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java [hunk 17]
```diff
@@ -305,6 +305,31 @@ public void execute() {
...
+                    errStream.println("\t bal build " + packageName);
+                    break;
+
+                case "lib":
+                    errStream.println("Execute the below command to create the Ballerina archive:");
+                    errStream.println("\t bal pack " + packageName);
+                    break;
+
+                default:
+                    errStream.println("Execute the below command to run the program:");
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java:328]:
```suggestion
                    errStream.println("\tExecute the below command to run the program:");
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PackCommand.java [hunk 18]
```diff
@@ -265,6 +266,10 @@ public void execute() {
                 .addTask(new DumpBuildTimeTask(outStream), !project.buildOptions().dumpBuildTime())
                 .build();
 
+                outStream.println("What's Next");
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PackCommand.java:269]:
```suggestion
                outStream.println("\nWhat's Next");
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PackCommand.java [hunk 19]
```diff
@@ -265,6 +266,10 @@ public void execute() {
...
+                outStream.println("Execute the below command to publish the package to Ballerina Central.");
+                outStream.println("\t bal push ");
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PackCommand.java:271]:
```suggestion
                outStream.println("\tbal push ");
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PackCommand.java [hunk 20]
```diff
@@ -265,6 +266,10 @@ public void execute() {
                 .addTask(new DumpBuildTimeTask(outStream), !project.buildOptions().dumpBuildTime())
                 .build();
 
+                outStream.println("What's Next");
+                outStream.println("Execute the below command to publish the package to Ballerina Central.");
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PackCommand.java:270]:
```suggestion
                outStream.println("\tExecute the below command to publish the package to Ballerina Central.");
```

File: cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/NewCommandTest.java [hunk 21]
```diff
@@ -198,6 +198,11 @@ public void testNewCommandInExistingDirectory() throws IOException {
                 """;
         Assert.assertEquals(gitignoreContent.trim(), expectedGitignoreContent.trim());
         Assert.assertTrue(readOutput().contains("Created new package"));
+        Assert.assertTrue(output.contains("What's next"));
+        Assert.assertTrue(output.contains("Execute the below command to run the program:"));
+        Assert.assertTrue(output.contains("\t bal run " + name));
```
Comment (azinneera) [inline cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/NewCommandTest.java:203]:
```suggestion
        Assert.assertTrue(output.contains("\tbal run " + name));
```

File: cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/NewCommandTest.java [hunk 22]
```diff
@@ -198,6 +198,11 @@ public void testNewCommandInExistingDirectory() throws IOException {
...
+        Assert.assertTrue(output.contains("Else, execute the below command to build the executable JAR file:"));
+        Assert.assertTrue(output.contains("\t bal build " + name));
```
Comment (azinneera) [inline cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/NewCommandTest.java:205]:
```suggestion
        Assert.assertTrue(output.contains("\tbal build " + name));
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java [hunk 23]
```diff
@@ -304,6 +304,13 @@ public void execute() {
...
+
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java:313]: @tanbirali you can use the existing APIs to get the target directory and the package name. Please replace the code L307-L312 as follows:
```java
Path targetDirPath = project.targetDir();
        String packageName = project.currentPackage().packageName().toString();
        String jarFileName = packageName + ProjectConstants.BLANG_COMPILED_JAR_EXT;
        Path jarFilePath = Paths.get(System.getProperty(ProjectConstants.USER_DIR)).relativize(
                targetDirPath.resolve("bin").resolve(jarFileName));
        outStream.println("\nWhat's next?");
        outStream.println("\tExecute java -jar " + jarFilePath + " to run the program");
```

File: cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/NewCommandTest.java [hunk 24]
```diff
@@ -198,6 +198,11 @@ public void testNewCommandInExistingDirectory() throws IOException {
                 """;
         Assert.assertEquals(gitignoreContent.trim(), expectedGitignoreContent.trim());
         Assert.assertTrue(readOutput().contains("Created new package"));
+        Assert.assertTrue(output.contains("What's next"));
```
Comment (azinneera) [inline cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/NewCommandTest.java:201]: `output` must be declared first. Please add the following line before L201:
```java
String output = readOutput();
```

**Comment by azinneera** [comment]
You would encounter test failures in the `ballerina-cli` with these changes. Please change the output `.txt` files that are in https://github.com/ballerina-platform/ballerina-lang/blob/8ef9be112ff1bb02d8f647fb2aadced794417941/cli/ballerina-cli/src/test/resources/test-resources/command-outputs/ to reflect the changes.

E.g.

https://github.com/ballerina-platform/ballerina-lang/blob/8ef9be112ff1bb02d8f647fb2aadced794417941/cli/ballerina-cli/src/test/resources/test-resources/command-outputs/unix/build-empty-project-with-build-tools.txt#L8-L9 

Change the above file as. Applies to other output files as well:
https://github.com/ballerina-platform/ballerina-lang/blob/9276d9014adede8c8010f6b13eaf804371225fd7/cli/ballerina-cli/src/test/resources/test-resources/command-outputs/unix/build-empty-project-with-build-tools.txt#L8-L12

**Pull request merged**